pkg/rc/replication_controller_test.go

```diff
@@ -237,3 +237,39 @@ func TestUnschedule(t *testing.T) {
 	Assert(t).AreEqual(len(scheduled), 0, "expected a pod to have been unlabeled")
 	Assert(t).AreEqual(len(kp.manifests), 0, "expected manifest to have been unscheduled")
 }
+
+func TestPreferUnscheduleIneligible(t *testing.T) {
+	rcStore, kp, applicator, rc := setup(t)
+	err := applicator.SetLabel(labels.NODE, "node1", "nodeQuality", "good")
+	Assert(t).IsNil(err, "expected no error labeling node1")
+	err = applicator.SetLabel(labels.NODE, "node2", "nodeQuality", "good")
+	Assert(t).IsNil(err, "expected no error labeling node2")
+	err = applicator.SetLabel(labels.NODE, "node3", "nodeQuality", "good")
+	Assert(t).IsNil(err, "expected no error labeling node3")
+
+	quit := make(chan struct{})
+	rc.WatchDesires(quit)
+
+	rcStore.SetDesiredReplicas(rc.ID(), 3)
+	numNodes := waitForNodes(t, rc, 3)
+	Assert(t).AreEqual(numNodes, 3, "took too long to schedule")
+
+	scheduled := scheduledPods(t, applicator)
+	Assert(t).AreEqual(len(scheduled), 3, "expected a pod to have been labeled")
+	Assert(t).AreEqual(len(kp.manifests), 3, "expected a manifest to have been scheduled")
+
+	// Make node2 ineligible, so that it will be preferred for unscheduling
```

Comment:
yeah I think it's inevitable that bad code might get lucky. If it makes you feel better I marked node3 as bad once and my test failed